specs/mcpfile-schema-0.1.0.json

```
@@ -0,0 +1,477 @@
+  "$id": "https://github.com/genmcp/gen-mcp/pkg/mcpfile/mcp-file",
+  "$ref": "#/$defs/MCPFile",
+  "$defs": {
+    "AuthConfig": {
+      "properties": {
+        "authorizationServers": {
+          "items": {
+            "type": "string"
+          },
+          "type": "array",
+          "description": "List of authorization server URLs for token validation."
+        },
+        "jwksUri": {
+          "type": "string",
+          "description": "URI for the JSON Web Key Set (JWKS) used for token verification."
+        }
+      },
+      "additionalProperties": false,
+      "type": "object",
+      "description": "AuthConfig defines OAuth 2.0 authorization settings."
+    },
+    "MCPFile": {
+      "properties": {
+        "mcpFileVersion": {
+          "type": "string",
+          "description": "Version of the MCP file format."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the MCP server."
+        },
+        "version": {
+          "type": "string",
+          "description": "Semantic version of the MCP server."
+        },
+        "runtime": {
+          "$ref": "#/$defs/ServerRuntime",
+          "description": "Runtime configuration for the MCP server."
+        },
+        "tools": {
+          "items": {
+            "$ref": "#/$defs/Tool"
+          },
+          "type": "array",
+          "description": "List of tools provided by the server."
+        },
+        "prompts": {
+          "items": {
+            "$ref": "#/$defs/Prompt"
+          },
+          "type": "array",
+          "description": "List of prompts provided by the server."
+        }
+      },
+      "additionalProperties": false,
+      "type": "object",
+      "required": [
+        "mcpFileVersion",
+        "name",
+        "version"
+      ],
+      "description": "MCPFile is the root structure of an MCP configuration file."
+    },
+    "Prompt": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique identifier for the prompt."
+        },
+        "title": {
+          "type": "string",
+          "description": "Human-readable title for display purposes."
+        },
+        "description": {
+          "type": "string",
+          "description": "Detailed description of what the prompt does."
+        },
+        "arguments": {
+          "items": {
+            "$ref": "#/$defs/PromptArgument"
+          },
+          "type": "array",
+          "description": "List of template arguments for the prompt."
+        },
+        "inputSchema": {
+          "$ref": "#/$defs/Schema",
+          "description": "Schema describing input parameters."
+        },
+        "outputSchema": {
+          "$ref": "#/$defs/Schema",
+          "description": "Optional schema describing prompt output."
+        },
+        "invocation": {
+          "description": "Object describing how to invoke the prompt."
+        },
```

@aliok for the `invocation` section, it seems to not be able to pick up on the possible sub-schemas for those blocks. IMO, this is something that will take a bit of work to get complete, but eventually this should look something like:

```suggestion
        "invocation": {
          "description": "Object describing how to invoke the prompt.",
          "oneOf": [
            { "$ref": "http-invocation-schema.json" },
            { "$ref": "cli-invocation-schema.json" }
          ]
        },
```

This would likely required adding something like a `JSONSchema` method to one of the invocation interfaces (for example https://github.com/genmcp/gen-mcp/blob/266f68accf04025196f9a8aefcfa977f6c460008/pkg/invocation/types.go#L31-L34), and then using that (plus maybe the "registration" methods we have in the invocation package), to:
1. Generate a json schema for each invocation type
2. Reference those files within this one

---

Yeah, I will see what I can do.
https://github.com/invopop/jsonschema/?tab=readme-ov-file#custom-type-definitions defines some ways of achieving this.

---

Thanks!!